deeplearning4j-core/src/test/java/org/deeplearning4j/nn/transferlearning/TransferLearningComplex.java

```diff
@@ -0,0 +1,222 @@
+        /*
+                        inCentre        inRight
+                            |               |
+                        denseCentre0        |
+                            |               |
+             ---------  denseCentre1     denseRight0
+            /               |               |
+        subsetLeft(0-3) denseCentre2 ----mergeRight
+          |                 |               |
+     denseLeft0         denseCentre3       denseRight1
+          |                 |                |
+     outLeft             outCentre         outRight
+
+         */
+
+        ComputationGraphConfiguration conf
+                = overallConf.graphBuilder()
+                        .addInputs("inCentre","inRight")
+                        .addLayer("denseCentre0", new DenseLayer.Builder().nIn(10).nOut(9).build(),"inCentre")
+                        .addLayer("denseCentre1", new DenseLayer.Builder().nIn(9).nOut(8).build(),"denseCentre0")
+                        .addLayer("denseCentre2", new DenseLayer.Builder().nIn(8).nOut(7).build(),"denseCentre1")
+                        .addLayer("denseCentre3", new DenseLayer.Builder().nIn(7).nOut(7).build(),"denseCentre2")
+                        .addLayer("outCentre", new OutputLayer.Builder(LossFunctions.LossFunction.MSE).nIn(7).nOut(4).build(),"denseCentre3")
+                        .addVertex("subsetLeft", new SubsetVertex(0,3),"denseCentre1")
+                        .addLayer("denseLeft0", new DenseLayer.Builder().nIn(4).nOut(5).build(),"subsetLeft")
+                        .addLayer("outLeft", new OutputLayer.Builder(LossFunctions.LossFunction.MSE).nIn(5).nOut(6).build(),"denseLeft0")
+                        .addLayer("denseRight0", new DenseLayer.Builder().nIn(2).nOut(3).build(),"inRight")
+                        .addVertex("mergeRight", new MergeVertex(),"denseCentre2","denseRight0")
+                        .addLayer("denseRight1", new DenseLayer.Builder().nIn(10).nOut(5).build(),"mergeRight")
+                        .addLayer("outRight", new OutputLayer.Builder(LossFunctions.LossFunction.MSE).nIn(5).nOut(5).build(),"denseRight1")
+                        .setOutputs("outLeft","outCentre","outRight")
+                        .build();
+
+        ComputationGraph modelToTune = new ComputationGraph(conf);
+        modelToTune.init();
+
+        INDArray [] features = new INDArray[] {Nd4j.rand(10,10),Nd4j.rand(10,2)};
+        INDArray [] labels = new INDArray[] {Nd4j.rand(10,6),Nd4j.rand(10,4),Nd4j.rand(10,5)};
+
+        //Instead of manually setting these three vertices - the comp graph .setFeatureExtractor should have a way of saying freeze from inCentre to denseCentre2
+        //and should trace the path from inCentre to denseCenter2 and freeze only those (not freeze the mergeRight etc etc)
+        modelToTune.getVertex("denseCentre0").setLayerAsFrozen();
+        modelToTune.getVertex("denseCentre1").setLayerAsFrozen();
```

Comment:
A user should not have to do this manually. The user sets this in the builder with .setFeatureExtractor...